### ISSUE TITLE
Cannot get component property

### ISSUE THREAD

**Issue by matfer** [abhinaba-ghosh/cypress-react-selector#123]
I have a date picker component that has no text input, only various buttons, so it would be difficult to set the date in tests by simulating clicks. So I would like to get the function stored in its onDateChange property to call it directly. However I cannot access the property.

With this
`cy.react('DateInput').eq(0).getProps('onDateChange').then(onDateChange => onDateChange(new Date(2021, 4, 4)))`
I get the error `Previous subject found null. getProps() is a child command. Use with cy.getReact()`

While with this
`cy.getReact('DateInput').eq(0).getProps('onDateChange').then(onDateChange => onDateChange(new Date(2021, 4, 4)))`
I get the error
```
cy.eq() failed because it requires a DOM element.

The subject received was:

  > [Object{5}, Object{5}]

The previous command that ran was:

  > cy.then()

All 2 subject validations failed on this subject.
```

**Comment by abhinaba-ghosh** [comment]
Hi @matfer , Thanks for raising this. According to [this](https://github.com/abhinaba-ghosh/cypress-react-selector#find-element-by-react-component)

`cy.react` returns a DOM element where `cy.getReact` retruns a [resq node](https://github.com/abhinaba-ghosh/cypress-react-selector/blob/04542b9a528406c596eaba7fe13326457019c00a/index.d.ts#L4). Rightfully, `eq()` is not supported with `getReact` as it doesn't returns a DOM elemnt. However, `eq` is supported with `react()` call, as it retuns DOM element(s). But, `getProps()/getState()` is not supported with `getReact()`. 

Now, to be able use `getProps()` you need to use it only with `getReact()`. You construct the `getReact` call in such a way so that it returns only one `resq node`. 

something like, 

```js
cy.react('DateInput', {uniqueProp: propValue}).getProps('onDateChange')
```

Try the above code, if not works, raise a feature request to fetch indexed nodes from `getReact()`. 

Thanks.

**Comment by m4tf3r** [comment]
Thank you, I will try as soon as I can. However, I often have sequences of those date picker components, so I would like to select them by index. Is there some alternative to eq()?

**Comment by abhinaba-ghosh** [comment]
Hi @matfer , This is addressed in `v2.3.0`. You can fetch elements with below commands case by case basis, 


1. `cy.react` returns DOM element, so you can fetch the indexed node by [.eq(index)](https://docs.cypress.io/api/commands/eq.html), like:

```js
cy.react('MyComponent').eq(0).click();
```

2. `cy.getReact()` return RESQ node, so you can't fetch it through `.eq()`. You need to use `.nthNode(index)`, like:

```js
cy.getReact('MyComponent')
  .nthNode(0)
  .getProps('name')
  .should('eq', 'First Item');
```
Give it a try and let me know.

If this library helps your development journey and want to get the improvement going, feel free to [sponsor](https://www.buymeacoffee.com/abhinabaghosh) the project 😄

**Comment by matfer** [comment]
I tried with 2.3.0 and 2.3.1 but I get this error:
```
Error: Webpack Compilation Error
./node_modules/cypress-react-selector/src/reactHandler.js 124:62
Module parse failed: Unexpected token (124:62)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
|             value = Cypress.$(value);
|           }
>           return cy.verifyUpcomingAssertions(value, reactOpts?.options, {
|             onRetry: resolveValue,
|           });
 @ ./node_modules/cypress-react-selector/index.js 8:4-33
 @ ./cypress/support/index.js
```
Looks like it is complaining about my ts file. It works fine if I go back to 2.2.2.

**Comment by m4tf3r** [comment]
It is shown in the browser when I launch my Cypress test.

**Comment by abhinaba-ghosh** [comment]
Hi @matfer , can you try `v2.3.2` and let us know the outcome?

**Comment by matfer** [comment]
It works, thank you! However the index parameter of nthNode has ts type string, is this intentional?

**Comment by abhinaba-ghosh** [comment]
Nice catch @m4tf3r , would you mind raising a small PR for [this](https://github.com/abhinaba-ghosh/cypress-react-selector/blob/3aaf3934e6e7cc2d31e79455e6dca548c470a221/index.d.ts#L99)

**Comment by matfer** [comment]
Done! #130

**Comment by abhinaba-ghosh** [comment]
> Done! #130

@m4tf3r available in `v2.3.3`. Please verify.
